app/services/arcgis_api/geocoder.rb

```diff
@@ -40,28 +44,50 @@ def suggest(text)
     # @param magic_key [String] a magic key value from a previous call to the #suggest method
     # @return [Array<AddressCandidate>] AddressCandidates
     def find_address_candidates(magic_key)
-      url = "#{root_url}/findAddressCandidates"
+      url = "#{root_url}/servernh/rest/services/GSA/USA/GeocodeServer/findAddressCandidates"
       params = {
         magicKey: magic_key,
         outFields: 'StAddr,City,RegionAbbr,Postal',
         **COMMON_DEFAULT_PARAMETERS,
       }
 
       parse_address_candidates(
-        faraday.get(url, params) do |req|
+        faraday.get(url, params, dynamic_headers) do |req|
           req.options.context = { service_name: 'arcgis_geocoder_find_address_candidates' }
         end.body,
       )
     end
 
+    # Makes a request to retrieve a new token
+    # it expires (1 hour minutes).
+    # @return [String] the token
+    def retrieve_token!
+      token_body = Rails.cache.read(API_TOKEN_CACHE_KEY)
+
+      if token_body.blank?
+        token_body = request_token
+        Rails.cache.write(
+          API_TOKEN_CACHE_KEY, token_body['token'],
+          expires_at: Time.zone.at(token_body['expires'])
+        )
+      end
```

Comment:
There's a mismatch here between what we're saving and retrieving. This is expecting to retrieve a body, but saves only the token.

Reply:
I saw a test was failing and assumed it was related. If it wasn't, I think it'd be good to have some regression coverage for that.

Reply:
In your example, I think the absence of an `expires_at` is causing it to not persist at all.

```
[1] pry(main)> Rails.cache.write('asdf', 'hi', expires_at: Time.zone.now + 1.hour)
=> true
[2] pry(main)> Rails.cache.read('asdf')
=> "hi"
```